docs/active_storage_migration.md

```diff
@@ -0,0 +1,387 @@
+   document is different from the previous.
+
+### Rollback
+
+**Note 1:** This rollback should not be attempted after PR #3722 has been deployed
+(final step, above) as this will delete assets from Active Storage that do not
+exist in Paperclip.
+
+**Note 2:** Using `destroy` or `destroy_all` on `ActiveStorage::Attachment` may
+delete the actual files so `delete` is used instead, even though it is more
+involved.
+
+1) Get a list of stats reports records in Active Storage:
+
+```ruby
+rails> attachments = ActiveStorage::Attachment.where(record_type: 'Stats::StatsReport')
+```
+
+2) Record the list of blobs for these attachments:
+
+```ruby
+rails> blobs = ActiveStorage::Blob.where(id: attachments.pluck(:blob_id))
+```
+
+3) Remove attachment and blob records:
+
+```ruby
+rails> attachments.delete_all
+rails> blobs.delete_all
+```
+
+4) Delete checksums
+
+```ruby
+rails> Stats::StatsReport.update_all(as_document_checksum: nil)
+```
+
+## Messages
+
+1) Generate checksums:
+
+```bash
+$ bundle exec rails 'storage:calculate_checksums[messages]'
+```
+
+2) Ensure that the checksums have been created correctly:
+
+```ruby
+rails> ms = Message.all
+
+# Total number of messages
+rails> ms.count
+
+# Number of messages with nil checksum
+rails> ms.where(as_attachment_checksum: nil).count
+# Should be zero
+
+# Number of attachments with a valid checksum
+# A valid checksum ends with '==', such as '4+g5Nmjax5aCg3HtRmCO5Q=='
+rails> ms.select { |m| m.as_attachment_checksum&.match(/==$/) }.count
+# Should be the same as ms.count, above
+```
+
+3) Migrate assets details from Paperclip to Active Storage:
+
+```bash
+$ bundle exec rails 'storage:migrate[messages]'
+```
+
+4) Confirm that all messages have been migrated:
+
+```ruby
+# List of Paperclip filenames
+rails> pc_filenames = Message.pluck(:attachment_file_name).sort
+
+# List of Active Storage filenames
+rails> as_filenames = ActiveStorage::Attachment.where(record_type: 'Message').map(&:filename).map(&:to_s).sort
+
+rails> pc_filenames == as_filenames
+# => true inducates that the migration is successful
+# => false indicates that the migraiton has failed
+```
+
+5) Test that a sample message attachment can be downloaded via Active Storage (this will only work when S3 is used for storage):
+
+```ruby
+# Get a download url for a sample attachment
+rails> ActiveStorage::Attachment.where(record_type: 'Message').sample.service_url
+# => "https://cloud-platform-..."
+# Paste this url into a browser and confirm that it can be downloaded
+# Note that this url will expire in 5 minutes
+```
+
+6) (Final step) Activate Active Storage for stats reports by deploying PR #3735 (CBO-1692)
+
+### Testing
+
+1) Find an existing case with a message attachment. Confirm that the attachment
+   can be downloaded correctly.
+2) Add a new message to a case and add an attachment. Confirm that the
+   attachment can be downloaded correctly.
+
+### Rollback
```

Comment:
[comment as for stats_reports](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3713/files#r581210705)